verde/coordinates.py

```diff
@@ -702,3 +728,99 @@ def block_split(coordinates, spacing, adjust="spacing", region=None):
+    """
+    Modify geographic region boundaries to ensure continuity around the globe.
+
+    Longitudinal boundaries of the region are moved to the `[0, 360)` or `[-180, 180)`
+    degrees interval depending which one is better suited for that specific region.
+
+    Parameters
+    ----------
+    region : list or array
+        List or array containing the boundary coordinates `w, `e`, `s`, `n` of the
+        region in degrees.
+    coordinates : list or array (optional)
+        Extra set of geographic coordinates that will be moved to the same degrees
+        interval as the one of the modified region.
+
+    Returns
+    -------
+    modified_region : array
+        List containing the modified boundary coordinates `w, `e`, `s`, `n` of the
+        region.
+    modified_coordinates : array (optional)
+        Modified set of extra geographic coordinates.
+
+    Examples
+    --------
+
+    >>> from verde import latlon_continuity
+    >>> # Modify region with west > east
+    >>> w, e, s, n = 350, 10, -10, 10
+    >>> print(latlon_continuity([w, e, s, n]))
+    [-10  10 -10  10]
+    >>> # Modify region and extra coordinates
+    >>> from verde import grid_coordinates
+    >>> region = [-70, -60, -40, -30]
+    >>> coordinates = grid_coordinates([270, 320, -50, -20], spacing=5)
+    >>> region, [longitude, latitude] = latlon_continuity(region, coordinates)
+    >>> print(region)
+    [290 300 -40 -30]
+    >>> print(longitude.min(), longitude.max())
+    270.0 320.0
+    >>> # Another example
+    >>> region = [-20, 20, -20, 20]
+    >>> coordinates = grid_coordinates([0, 350, -90, 90], spacing=10)
+    >>> region, [longitude, latitude] = latlon_continuity(region, coordinates)
+    >>> print(region)
+    [-20  20 -20  20]
+    >>> print(longitude.min(), longitude.max())
+    -180.0 170.0
+    """
+    # Get longitudinal boundaries and check region
+    w, e, s, n = region[:4]
+    check_region([w, e, s, n], latlon=True)
+    # Check if region is defined all around the globe
+    all_globe = np.allclose(abs(e - w), 360)
+    # Move coordinates to [0, 360)
+    interval_360 = True
+    w = w % 360
+    e = e % 360
+    # Move west=0 and east=360 if region longitudes goes all around the globe
+    if all_globe:
+        w, e = 0, 360
+    # Check if the [-180, 180) interval is better suited
+    if w > e:
+        interval_360 = False
+        e = ((e + 180) % 360) - 180
+        w = ((w + 180) % 360) - 180
+    region = np.array(region)
+    region[:2] = w, e
+    # Modify extra coordinates if passed
+    if coordinates:
+        _check_geographic_coordinates(coordinates)
+        longitude = coordinates[0]
+        if interval_360:
+            longitude = longitude % 360
+        else:
+            longitude = ((longitude + 180) % 360) - 180
+        coordinates = np.array(coordinates)
+        coordinates[0] = longitude
+        return region, coordinates
+    return region
+
+
+def _check_geographic_coordinates(coordinates):
```

Comment:
I don't think this function looks bad. Seems reasonable to me and does what it should.

Reply:
:+1: